lib/utilities.bash

```diff
@@ -139,25 +139,47 @@ function _bash-it-clean-component-cache() {
 
 # Returns an array of items within each compoenent.
 function _bash-it-component-list() {
-	local IFS=$'\n' component="$1"
-	_bash-it-component-help "${component}" | awk '{print $1}' | sort -u
+	local all component_dir
+
+    _bash-it-component-pluralize "$1" pluralized_component
+    component_dir="${pluralized_component/completions/completion}"
+    all=("$BASH_IT/$component_dir/available/"*.bash); all=("${all[@]##*/}"); all=("${all[@]%.*.bash}")
+    builtin printf '%s\n' "${all[@]}"
 }
 
 function _bash-it-component-list-matching() {
 	local component="$1"
 	shift
 	local term="$1"
-	_bash-it-component-help "${component}" | _bash-it-egrep -- "${term}" | awk '{print $1}' | sort -u
+	_bash-it-component-list "${component}" | _bash-it-egrep -- "${term}"
 }
 
 function _bash-it-component-list-enabled() {
-	local IFS=$'\n' component="$1"
-	_bash-it-component-help "${component}" | _bash-it-egrep '\[x\]' | awk '{print $1}' | sort -u
+	local component="${1/alias/aliases}" enabled
+
+    _bash-it-component-pluralize "$1" pluralized_component
+    component_dir="${pluralized_component/completions/completion}"
+    [[ -d "$BASH_IT/$component_dir/enabled" ]] &&
+    enabled=($(compgen -G "$BASH_IT/$component_dir/enabled/*.$component.bash")) ||
+    enabled=($(compgen -G "$BASH_IT/enabled/*.$component.bash"))
```

Comment:
@tsiflimagas, Globbing with `compgen` should be avoided; it's rarely correct without a lot of work.

(To be correct, it would require using `IFS=$'\n' read -d '' -ra some_var < <(compgen -G "/the/stuff/*")` and possibly even `printf '%q' "${some_var[@]}"`...)

Reply:
Oh, haha, you picked that? 😅 I had made some very sketchy changes, which were half broken, but then I got in lack of time, so I pushed them to continue with those later. I see you reworked them to perfection though, so it's great! Also, unsetting array subscripts, damn, I never thought of that. It's perfect. I always did `arr="${arr//i}"`, which would leave an empty member.
About `compgen`, thanks for indicating that, it's important to me when I learn something from someone who's better at it. Usually, when I'm sure that the files I'm working with won't have something weird in their names, I let myself be more incautious. But yeah, it's better to always avoid bad practices (especially in bash it seems to be too easy to slip).